### DISCUSSION TITLE
`wxt zip` error build.

### DISCUSSION THREAD

**Discussion by xunziheng** [wxt-dev/wxt#756] [Q&A] [answered]
I'm having a strange problem：

The program works fine while developing.

But when I executed `pnpm zip` and dragged the packaged zip file into the chrome extension manager, something strange happened: 
The `tab` object printed by the following code did not have `url`, so `tab.url` was undefined. I am not sure whether there are other similar problems, because this problem prevented my program from working in the production environment before publish.

following code:
```js
browser.tabs.onActivated.addListener(async (activeInfo) => {
      const tab = await browser.tabs.get(activeInfo.tabId);
      console.log('tab1: ', tab)
      checkState(tab.url, tab.id);  // tab.url is undefined, but it worked fine while developing.
});
```

<img width="691" alt="image" src="https://github.com/wxt-dev/wxt/assets/35691153/d910a24c-8b90-4cf8-bd50-14235ef94b85">

**Comment by aklinker1** [comment, marked as answer]
Have you requested the `tabs` permission? Those APIs are only available if you've added that permission to the manifest.

https://developer.chrome.com/docs/extensions/reference/api/tabs#perms

Dev mode adds some, like tabs, so it can reload content scripts on change. But those permissions don't show up in production builds unless you explicitly add them.
